### PR TITLE
Add :focus-within support  

### PR DESCRIPTION
**Pull request by byteab**
- [x] web support
- [x] native support
- [x] CSS only support
- [x] fix `focusVisible` ( add missing doc, and fix CSS only mode )
- [x] `$group-` support
- [x] document and demo

### CODE REVIEW

**Comment by natew** [comment]
@byteab how close is this one?

**Comment by byteab** [comment]
> @byteab how close is this one?

just need to be added to doc

**Comment by byteab** [comment]
@natew it should be ready now

File: code/core/web/src/createComponent.tsx [hunk 1]
```diff
@@ -688,6 +689,11 @@ export function createComponent<
 
     const unPress = () => setStateShallow({ press: false, pressIn: false })
 
+    if (propsIn.focusWithinStyle) {
+      componentContext.onWithinFocus = () => setStateShallow({ focusWithin: true })
```
Comment (natew) [inline code/core/web/src/createComponent.tsx:693]:
```suggestion
      componentContext.onWithinFocus ||= () => setStateShallow({ focusWithin: true })
```

Comment (natew) [reply code/core/web/src/createComponent.tsx:693]: for both

File: code/core/web/src/helpers/pseudoDescriptors.ts [hunk 2]
```diff
@@ -20,6 +20,11 @@ export const pseudoDescriptorsBase = {
     name: 'focus',
     priority: 3,
   },
+  focusWithinStyle: {
```
Comment (natew) [inline code/core/web/src/helpers/pseudoDescriptors.ts:23]: what happens when you have both visible and within? they collide right?

Comment (byteab) [reply code/core/web/src/helpers/pseudoDescriptors.ts:23]: yeah both have same `specificity` so should respect order

**Pull request merged**